packages/patterns/src/types.js

```diff
@@ -492,12 +492,12 @@ export {};
 
 /**
  * @typedef {{
- * <M extends Record<any, MethodGuard>>(interfaceName: string,
- *             methodGuards: M,
- *             options?: {sloppy?: false}): InterfaceGuard<M>;
  * (interfaceName: string,
  *             methodGuards: any,
- *             options?: {sloppy?: true}): InterfaceGuard<any>;
+ *             options: {sloppy: true}): InterfaceGuard<Record<PropertyKey, MethodGuard>>;
+ * <M extends Record<PropertyKey, MethodGuard>>(interfaceName: string,
+ *             methodGuards: M,
+ *             options?: {sloppy?: boolean}): InterfaceGuard<M>;
  * }} MakeInterfaceGuard
  */
```

Comment:
This is quite hard to read; can it be improved by naming the alternatives? Untested:
```js
/**
 * @typedef {(interfaceName: string, methodGuards: object, options: {sloppy: true}) => InterfaceGuard<Record<PropertyKey, MethodGuard>>} SloppyInterfaceGuard
 */
/**
 * @typedef {<M extends Record<PropertyKey, MethodGuard>>(interfaceName: string, methodGuards: M, options?: {sloppy?: boolean}) => InterfaceGuard<M>} StrictInterfaceGuard
 */
/**
 * @typedef {SloppyMakeInterfaceGuard | StrictMakeInterfaceGuard} MakeInterfaceGuard
 */
```